lib/jekyll/commands/serve.rb

```diff
@@ -51,6 +107,70 @@ def process(opts)
           start_up_webrick(opts, destination)
         end
 
+        def shutdown
+          @server.shutdown if @running
+        end
+
+        private
+        def validate_options(opts)
+          if opts["livereload"]
+            if opts["detach"]
+              Jekyll.logger.abort_with "Error:",
+                "--detach and --livereload are mutually exclusive"
```

Comment:
I'd like to go out on a limb here and have the `--livereload` option take precedence over the `--detach` option so that if both are specified together, we simply don't detach.